src/wp-includes/class-wp-image-editor.php

```diff
@@ -469,21 +457,7 @@ public function generate_filename( $suffix = null, $dest_path = null, $extension
 			}
 		}
 
-		if ( empty( $suffix ) ) {
-			$suffix = '';
-		} else {
-			$suffix = "-{$suffix}";
-		}
-
-		// When the file extension being generated doesn't match the image file extension,
-		// add the extension to the suffix to ensure a unique file name. Prevents
-		// name conflicts when a single image type can have multiple extensions,
-		// eg. .jpg, .jpeg and .jpe are all valid JPEG extensions.
-		if ( ! empty( $extension ) && $extension !== $ext ) {
-			$suffix .= "-{$ext}";
-		}
```

Comment:
Didn't we say we should keep this part to ensure unique file names? Otherwise we would re-introduce that bug.

Reply:
@adamsilverstein Nevermind, I see this will be re-added via https://github.com/adamsilverstein/wordpress-develop-fork/pull/35.